src/Requests.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Biigle\Modules\Kpis;
+
+use Carbon\Carbon;
+use Illuminate\Support\Facades\DB;
+
+class Requests
+{
+    public static function save($visits, $actions)
+    {
+        DB::transaction(function () use ($visits, $actions) {
+            $today = Carbon::today()->toDateString();
+            DB::table('kpis_actions')->insert(['date' => $today, 'value' => $actions]);
+            DB::table('kpis_visits')->insert(['date' => $today, 'value' => $visits]);
+        });
+    }
+
+    public static function getActions($year, $month)
+    {
+        $date = Carbon::createFromDate($year, $month, 1)->endOfMonth()->toDateString();
+        $res =  count(DB::table('kpis_actions')->where('date', '=', $date)->pluck('value')) != 0 ?
+        DB::table('kpis_actions')->where('date', '=', $date)->pluck('value')[0] : 0;
```

Comment:
Always try to do as much work as possible directly in the DB. Also the query seems wrong as it returns only the actions of the last day of the month and not the whole month:

```suggestion
        $start = Carbon::createFromDate($year, $month, 1)->toDateString();
        $end = Carbon::createFromDate($year, $month, 1)->endOfMonth()->toDateString();
        $res = DB::table('kpis_actions')->whereBetween('date', [$start, $end])->sum('value');
```

Reply:
Please also add a test case that checks if `$end` is included or not. I'm not sure how `whereBetween()` behaves with these dates.

Reply:
Ah sorry. I read 'daily' in the issue but had monthly in my mind somehow.
But my [tests](https://github.com/biigle/kpis/blob/564c45a02a0a31571fe27b149600fd1d84ff2315/tests/RequestsTest.php#L45-L64) already use the last day of the month to check the values.

Reply:
No, the tests should be fine then!